Contact/Console/Command/ExportContacts.php

```diff
@@ -0,0 +1,78 @@
+<?php
+namespace Gorilla\Contact\Console\Command;
+
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Magento\Framework\App\Filesystem\DirectoryList;
+
+/**
+ * Class ExportContacts
+ */
+class ExportContacts extends Command
+{
+
+    protected $_directory;
+
+    protected $_contactFactory;
+
+    /**
+     * ExportContacts constructor.
+     *
+     * @param \Magento\Framework\Filesystem $filesystem
+     * @param \Gorilla\Contact\Model\ContactFactory $contactFactory
+     */
+    public function __construct(
+        \Magento\Framework\Filesystem $filesystem,
+        \Gorilla\Contact\Model\ContactFactory $contactFactory
+    ) {
+        parent::__construct();
+        $this->_directory = $filesystem->getDirectoryWrite(DirectoryList::VAR_DIR);;
+        $this->_contactFactory = $contactFactory;
+    }
+    /**
+     * @inheritDoc
+     */
+    protected function configure()
+    {
+        $this->setName('gorilla:export:contacts')
+            ->setDescription('Export Contacts');
+
+        parent::configure();
+    }
+
+    /**
+     * Execute the command
+     *
+     * @param InputInterface $input
+     * @param OutputInterface $output
+     *
+     * @return null|int
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $filePath = 'export/contacts-' . date('Y-m-d--H-i-s') . '.csv';
+        $output->writeln('<info>Exporting contacts to: ' . $filePath . '</info>');
+
+        $this->_directory->create('export');
+        $stream = $this->_directory->openFile($filePath, 'w+');
+        $stream->lock();
```

Comment:
Ideally the file should be unlocked or closed when we're done with it. Otherwise it will only be released once the PHP process is complete.